modules/nf-core/cellsnp/modea/tests/main.nf.test

```diff
@@ -0,0 +1,91 @@
+    tag "cellsnp"
+    tag "cellsnp/modea"
+    tag "samtools/index"
+
+    test("genotyping") {
+        setup { 
+            run("SAMTOOLS_INDEX") {
+                script "../../../samtools/index/main.nf"
+                process {
+                    """
+                    input[0] = [
+                    [ id:'sample1' ],
+                    file(params.modules_testdata_base_path + '/genomics/homo_sapiens/demultiplexing/chr21.bam', checkIfExists: true) ]
+                    """
+                }
+            }
+        }
+
+        when {
+            process {
+                """ 
+                input[0] = SAMTOOLS_INDEX.out.bai.collect{ meta, bai -> bai }.map{ 
+                            bai -> [[ id: 'sample1'], 
+                                    file(params.modules_testdata_base_path + '/genomics/homo_sapiens/demultiplexing/chr21.bam', checkIfExists: true),
+                                    bai,
+                                    file(params.modules_testdata_base_path + '/genomics/homo_sapiens/demultiplexing/donor_genotype_chr21.vcf', checkIfExists: true),
+                                    file(params.modules_testdata_base_path + '/genomics/homo_sapiens/demultiplexing/barcodes.tsv', checkIfExists: true)                                    ]}
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out.versions).match("versions") },
+	            { assert path(process.out.base.get(0).get(1)).exists() },
+                { assert snapshot(
+                    process.out.sample,
+                    process.out.allele_depth,
+                    process.out.depth_coverage,
+                    process.out.depth_other).match()
+                }    
+	        )
+        }
+
+    }
+
+    test("genotyping - stub") {
+
+        options "-stub"
+
+        when {
+            process {
+                """
+                
+                input[0] = [
+                    [ id:'sample1'], // meta map
+                    file(params.modules_testdata_base_path + '/genomics/homo_sapiens/demultiplexing/chr21.bam', checkIfExists: true),
+                    [],
+                    file(params.modules_testdata_base_path + '/genomics/homo_sapiens/demultiplexing/donor_genotype_chr21.vcf', checkIfExists: true),
+                    file(params.modules_testdata_base_path + '/genomics/homo_sapiens/demultiplexing/barcodes.tsv', checkIfExists: true)
+                    ]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+            { assert process.success },
+            { assert snapshot(
+                    process.out.sample,
+                    process.out.allele_depth,
+                    process.out.depth_coverage,
+                    process.out.depth_other).match("stub")
+            },
+	        { assert path(process.out.base.get(0).get(1)).exists() } )
```

Comment:
Stub test can just check `process.out`, as nothing should vary.

Reply:
Oh, true. But you should be able to get around that by swapping to  `echo "" | gzip > ${prefix}.base.vcf.gz`